cognite/neat/rules/models/_rules/_types.py

```diff
@@ -13,16 +57,126 @@
     BeforeValidator(lambda v: [entry.strip() for entry in v.split(",")] if isinstance(v, str) else v),
 ]
 
+Namespace_ = Annotated[
+    Namespace,
+    BeforeValidator(
+        lambda v: (
+            Namespace(TypeAdapter(HttpUrl).validate_python(v))
+            if v.endswith("#") or v.endswith("/")
+            else Namespace(TypeAdapter(HttpUrl).validate_python(f"{v}/"))
+        )
+    ),
+]
+
+Prefix = Annotated[
+    str,
+    StringConstraints(pattern=prefix_compliance_regex),
+    custom_error(
+        lambda field_name, error, value: exceptions.PrefixesRegexViolation(
+            [value], prefix_compliance_regex
+        ).to_pydantic_custom_error()
+    ),
+]
+
 ExternalID = Annotated[
     str,
     Field(min_items=1, max_items=255),
 ]
 
 Version = Annotated[
     str,
-    Field(
-        description="Data model version",
-        min_length=1,
-        max_length=43,
+    StringConstraints(pattern=version_compliance_regex),
+    custom_error(
+        lambda field_name, error, value: exceptions.VersionRegexViolation(
+            value, version_compliance_regex
+        ).to_pydantic_custom_error()
+    ),
+]
+
+
+def split_parent(value: str) -> list[ParentClass]:
+    if isinstance(value, str) and value:
+        parents = []
+        for v in value.replace(", ", ",").split(","):
+            if ENTITY_ID_REGEX_COMPILED.match(v) or VERSIONED_ENTITY_REGEX_COMPILED.match(v):
+                parents.append(ParentClass.from_string(entity_string=v))
+            else:
+                # if all fails defaults "neat" object which ends up being updated to proper
+                # prefix and version upon completion of Rules validation
+                parents.append(ParentClass(prefix="undefined", suffix=v, name=v))
+
+        return parents
+    else:
+        return None
+
+
+def check_parent(value: list[ParentClass]) -> list[ParentClass]:
+    if value:
+        if illegal_ids := [v for v in value if re.search(more_than_one_none_alphanumerics_regex, v.suffix)]:
+            raise exceptions.MoreThanOneNonAlphanumericCharacter(
+                "parent", ", ".join(illegal_ids)
+            ).to_pydantic_custom_error()
+        if illegal_ids := [v for v in value if not re.match(class_id_compliance_regex, v.suffix)]:
+            for v in illegal_ids:
+                print(v.id)
+            raise exceptions.ClassSheetParentClassIDRegexViolation(
+                illegal_ids, class_id_compliance_regex
+            ).to_pydantic_custom_error()
+    return value
+
+
+ParentClass_ = Annotated[
+    list[ParentClass] | None,
+    BeforeValidator(lambda v: split_parent(v)),
+    AfterValidator(lambda v: check_parent(v)),
```

Comment:
No-op lambda is no point :) 
```suggestion
    BeforeValidator(split_parent),
    AfterValidator(check_parent),
```